components/db/events.go

```diff
@@ -0,0 +1,284 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/sqlite"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	// Event timestamp in unix seconds.
+	ColumnTimestamp = "timestamp"
+
+	// event name
+	// e.g., "memory_oom", "memory_oom_kill_constraint", "memory_oom_cgroup", "memory_edac_correctable_errors".
+	ColumnName = "name"
+
+	// event type
+	// e.g., "Unknown", "Info", "Warning", "Critical", "Fatal".
+	ColumnType = "type"
+
+	// event extra info
+	// e.g.,
+	// data source: "dmesg", "nvml", "nvidia-smi".
+	// event target: "gpu_id", "gpu_uuid".
+	// log detail: "oom_reaper: reaped process 345646 (vector), now anon-rss:0kB, file-rss:0kB, shmem-rss:0".
+	ColumnExtraInfo = "extra_info"
+
+	// event suggested actions
+	// e.g., "reboot"
+	ColumnSuggestedActions = "suggested_actions"
+)
+
+type Event struct {
+	Timestamp        int64  `json:"timestamp"`
+	Name             string `json:"name"`
+	Type             string `json:"type"`
+	ExtraInfo        string `json:"extra_info"`
+	SuggestedActions string `json:"suggested_actions"`
+}
+
+type storeImpl struct {
+	table string
+	dbRW  *sql.DB
+	dbRO  *sql.DB
+}
+
+var (
+	ErrNoDBRWSet = errors.New("no writable db set")
+	ErrNoDBROSet = errors.New("no read-only db set")
+)
+
+// Creates a new DB instance with the table created.
+func NewStore(ctx context.Context, dbRW *sql.DB, dbRO *sql.DB, tableName string) (*storeImpl, error) {
+	if dbRW == nil {
+		return nil, ErrNoDBRWSet
+	}
+	if dbRO == nil {
+		return nil, ErrNoDBROSet
+	}
```

Comment:
if we do not allow them to be nil, what's the purpose of separating them? Can we use dbRW for both read and write?

Reply:
Oh we want to isolate reads since we have different open modes for write and reads.

Reply:
https://github.com/leptonai/gpud/blob/main/pkg/sqlite/sqlite.go#L34-L35

Reply:
i understand this but what's the impact of GPUd if we use dbRW for both read and write?

Reply:
`getEvents` (reads) will need to wait when write transactiosn ongoing (insert events).

Reply:
i don't think writes to table A will impact reads from table B... anyways we can keep it like this for now